core/src/Cuda/Kokkos_Cuda_Half.hpp

```diff
@@ -142,8 +142,8 @@ class alignas(2) half_t {
   half_t() : val(0.0F) {}
 
   // Copy constructors
-  KOKKOS_DEFAULTED_FUNCTION
-  half_t(const half_t&) noexcept = default;
+  KOKKOS_FUNCTION
+  half_t(const half_t& rhs) : val(rhs.val){};
```

Comment:
Did that have anything to do with the `noexcept` specification?
Consider guarding the code and please add a comment.

Reply:
ah shit it was guarded in my code first but the rebase was weird, gonna redo that.

Reply:
reintroduced the guard stuff